### PR TITLE
Add logging for registering stratifications and observations

### PR DESCRIPTION
**Pull request by rmudambi**
### Description

- *Category*: feature
- *JIRA issue*: [MIC-4228](https://jira.ihme.washington.edu/browse/MIC-4228)

Added logging to the results manager's registration methods.
This matches the usage in `ValuesManager`

### Testing
Ran unit tests and  ran a simple simulation and verified log messages appear

### CODE REVIEW

File: tests/framework/results/test_interface.py [hunk 1]
```diff
@@ -91,9 +91,13 @@ def test_register_observation(
...
 
 
-def test_register_observations():
+def test_register_observations(mocker):
     mgr = ResultsManager()
     interface = ResultsInterface(mgr)
+    builder = mocker.Mock()
+    builder.configuration.stratification.default = []
+    mgr.setup(builder)
+
```
Comment (mattkappel) [inline tests/framework/results/test_interface.py:100]: Why did you add this mock and setup() to these tests?

**Review by mattkappel — APPROVED** [review]

File: tests/framework/results/test_interface.py (hunk 1 above)
Comment (stevebachmeier) [reply tests/framework/results/test_interface.py:100]: Curious as well.

**Review by stevebachmeier — APPROVED** [review]

**Review by hussain-jafari — APPROVED** [review]

File: tests/framework/results/test_interface.py (hunk 1 above)
Comment (rmudambi) [reply tests/framework/results/test_interface.py:100]: The test fails without `setup` being run, as the `logger` doesn't get created without that. When `setup` is run it overrides the default stratification with a Mock object rather than the empty list it was before.
